.cbmc-batch/jobs/aws_ring_buffer_buf_belongs_to_pool/aws_ring_buffer_buf_belongs_to_pool_harness.c

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/common/byte_buf.h>
+#include <aws/common/ring_buffer.h>
+#include <proof_helpers/make_common_data_structures.h>
+#include <proof_helpers/proof_allocators.h>
+
+void aws_ring_buffer_buf_belongs_to_pool_harness() {
+    /* parameters */
+    struct aws_ring_buffer ring_buf;
+    size_t ring_buf_size;
+    struct aws_byte_buf buf;
+
+    /* assumptions */
+    ensure_ring_buffer_has_allocated_members(&ring_buf, ring_buf_size);
+    __CPROVER_assume(aws_ring_buffer_is_valid(&ring_buf));
+    ensure_byte_buf_has_allocated_buffer_member(&buf);
+    __CPROVER_assume(aws_byte_buf_is_valid(&buf));
+    __CPROVER_assume(__CPROVER_POINTER_OBJECT(buf.buffer) == __CPROVER_POINTER_OBJECT(ring_buf.allocation));
+
+    assert(aws_ring_buffer_buf_belongs_to_pool(&ring_buf, &buf));
```

Comment:
Is it worth doing
```c
if(nondet_bool())
  assert(aws_ring_buffer_buf_belongs_to_pool(&ring_buf, &buf));
else if(nondet_bool())
  aws_ring_buffer_buf_belongs_to_pool(NULL, &buf);
else if(nondet_bool())
  aws_ring_buffer_buf_belongs_to_pool(&ring_buf, NULL);
else if(nondet_bool())
  aws_ring_buffer_buf_belongs_to_pool(NULL, NULL);


/* assertions */
assert(aws_ring_buffer_is_valid(&ring_buf));
assert(aws_byte_buf_is_valid(&buf));
```
?

Reply:
Similarly to `acquire` this function has to be called with valid `buf` and `ring_buf`, and so it should never be called with them being NULL.